packages/core/src/fhir-deduplication/shared.ts

```diff
@@ -175,7 +175,7 @@ export function getDateFromResource<T extends Resource>(
     if (period.start) return getDateFromString(period.start, dateFormat);
   } else if ("effectivePeriod" in resource) {
     if (resource.effectivePeriod.start) {
-      return getDateFromString(resource.effectivePeriod.start);
+      return getDateFromString(resource.effectivePeriod.start, dateFormat);
```

Comment:
This is going to affect diagnostic reports directly because now the key will change from `date` to `datetime` - [context](https://www.notion.so/metriport/Data-Quality-v2-28a69479807a4f9b897921fd77e54d8f?pvs=4#8314bb4a76d54ef88cfe7973a6ee9105)